### ISSUE TITLE
Performance Issue: No output in StdOut 

### ISSUE THREAD

**Issue by yaomz16** [ArjunaCluster/ArjunaUsers#55]
Your Name: Archie
Your Andrew ID: mingzeya
Node(s) on which the problem occurred:c002
Expected Behavior: slurm writes the printed info to StdOut file
Observed Behavior: nothing printed (except two echos I set in .sh file) in StdOut
Location of Log file Showing the Error: /home/mingzeya/CGCNN_project/AbNN_benchmark/hform/case1/out_245.out
Location of Script showing [Minimum Working Example]: /home/mingzeya/CGCNN_project/AbNN_benchmark/hform/case1/TrainGPU.sh

I know StdOut files can be empty in the first ~2 hours, but it has been ~1 day, there's still nothing in the StdOut file (There are plenty of print function in my script  /home/mingzeya/CGCNN_project/AbNN_benchmark/hform/case1/main.py, the printed info should be appear in StdOut).

**Comment by abillscmu** [comment]
Try doing it interactively and narrow down the issue. I can't debug the script for you, you need to isolate the issue and let us know if there's an issue with the system.

**Comment by yaomz16** [comment]
Hi Alec, Thanks for the guidance. I tried to do it interactively, and found that the printed info can be printed normally on the screen:

"

(abnn) mingzeya@c002:~/CGCNN_project/AbNN_benchmark/hform/case1_INTERACTIVE    $
 python3 main.py --prop hform --epochs 400 --lr_milestones 50 100 200 --gamma 0.25 --normalizing_const_AO 10000.0 --lr_generator  0.001 --lr_conv 0.001 --print_freq 32 --csv_dir ../case1/csv_files_npz/ --npz_dir ../case1/npz_files/


*****


CUDA available? True


*****
Pretrained Generator Loaded.
# of training data:  1736
# of validation data:  372
# of test data:  372
Training Starts:
Epoch: [0][31/1736]	 Loss 0.9956 MAE 0.8443 RMSE 0.9978
Epoch: [0][63/1736]	 Loss 1.0715 MAE 0.6233 RMSE 1.0352
...

"

I run the job interactively under the directory /home/mingzeya/CGCNN_project/AbNN_benchmark/hform/case1__INTERACTIVE, with the command "srun -J tmpInteractive  -A venkvis_gpu -p gpu -n 4 -N 1 --mem=8000 --gres=gpu:2 --pty bash" to activate the interactive mode.

I expected that these lines (except the first line, that's typing the command into terminal) to be appeared in the StdOut file. According to my past experience, they would indeed appeared in the StdOut file (though they might be appeared a little bit late, say ~2hrs after the job began to run), but it is strange that there is still no printed info now (it has been ~1 day since the job began).

**Comment by yaomz16** [comment]
I also tried to write everything I printed in a stand-alone file (syn_output.txt), and this works well. You can find it at /home/mingzeya/CGCNN_project/AbNN_benchmark/hform/case2/syn_output.txt. However, the StdOut file (/home/mingzeya/CGCNN_project/AbNN_benchmark/hform/case2/out_499.out) is still emtpy. Therefore, I think this is a slurm issue.

**Comment by abillscmu** [comment]
I cannot reproduce this. Please provide an MWE if you would like me to help you more.

**Comment by yaomz16** [comment]
Hi Alec, thanks for inspecting! MWE is located at /home/mingzeya/CGCNN_project/AbNN_benchmark/bandgap/MWE.
To submit the job to batch: sbatch TrainGPU.sh
There will be a file called syn_output.txt in the MWE directory which contains everything I printed in train.py
I expected the out_JOBID.out (StdOut file) to be like this (simply two line, without anything else)
"
Job started on c0** at $$TIME$$
/home/mingzeya/CGCNN_project/AbNN_benchmark/bandgap/MWE
"

**Comment by abillscmu** [comment]
That's not an MWE. An MWE is a very simple script which reproduces the issue. Please simplify your script and job as much as possible while still reproducing the issue.

https://en.wikipedia.org/wiki/Minimal_working_example

**Comment by abillscmu** [comment]
updates? Or can I close this?

**Comment by yaomz16** [comment]
Oops... I forgot to close this. I found that there will be output after the job is completed. It's just delayed for a very long time. I  resolved this problem by adding a manual output to a stand-alone txt file.
